content/Software and Downloads/Upload/Error-avrdude-when-uploading.md

```diff
@@ -2,58 +2,87 @@
 title: "'Error: avrdude' when uploading"
 ---
 
-The tool **avrdude** is used to upload sketches to the Arduino boards which use microcontrollers of the AVR architecture. When uploading code to your board in the IDE, avrdude may give an error. Some of the most common errors are listed below.
+The tool **avrdude** is used to upload sketches to the Arduino boards which use microcontrollers of the AVR architecture. When uploading code to your board in the IDE, avrdude may give an error.
 
-* [`avrdude: stk500_recv(): programmer is not responding`](#avrdude-stk500_recv-and-stk500_getsync)
-* [`avrdude: stk500_getsync() attempt X of 10: not in sync`](#avrdude-stk500_recv-and-stk500_getsync)
-* [`avrdude: stk500v2_getsync(): timeout communicating with programmer`](#avrdude-stk500_recv-and-stk500_getsync)
-* [`avrdude: butterfly_recv(): programmer is not responding`](#avrdude-butterfly_recv)
-* [`avrdude: ser_open(): can't open device "portName": no such file or directory`](#avrdude-ser_open-cant-open-device-no-such-file-or-directory)
-* [`avrdude: ser_open(): can't open device "portName": the system cannot find the file specified.`](#check-the-following)
-* [`avrdude: ser_open(): can't open device "portName": access is denied.`](#check-the-following)
-* [`avrdude: ser_open(): can't open device "portName": permission denied`](#check-the-following)
-* [`avrdude: ser_open(): can't open device "portName": the semaphore timeout period has expired.`](#check-the-following)
-* [`avrdude: ser_open(): can't set com-state for "portName"`](#check-the-following)
+Most of these errors can be resolved by following this [general troubleshooting checklist](#check-the-following).
 
-> Text following `avrdude: ser_open():` can appear in a different language if your system language is not set to English.
+You can also read about these specific errors:
 
-While these errors can be somewhat ambiguous, they are most often caused by failing or incorrectly configured hardware. It is likely that the error can be solved by following this troubleshooting procedure.
+* [avrdude: stk500_recv(): programmer is not responding
+   avrdude: stk500_getsync() attempt X of 10: not in sync](#avrdude-stk500_recv-and-stk500_getsync)
+* [avrdude: stk500v2_ReceiveMessage(): timeout
+  avrdude: stk500v2_getsync(): timeout communicating with programmer](#avrdude-stk500_recv-and-stk500_getsync)
+* [avrdude: ser_open(): can't open device "portName": no such file or directory](#avrdude-ser_open-cant-open-device-no-such-file-or-directory)
```

Comment:
The commented out "TODO" list below needs to be implemented for the links to work